homeassistant/components/ovo_energy/__init__.py

```diff
@@ -59,7 +68,17 @@ async def async_update_data() -> OVODailyUsage:
                 raise UpdateFailed(exception) from exception
             if not authenticated:
                 raise ConfigEntryAuthFailed("Not authenticated with OVO Energy")
-            return await client.get_daily_usage(datetime.utcnow().strftime("%Y-%m"))
+
+            daily_usage = await client.get_daily_usage(
+                datetime.utcnow().strftime("%Y-%m")
+            )
+            if daily_usage is None:
+                raise UpdateFailed("No daily usage data found")
```

Comment:
Is it better to still show the plan even if the daily usage data is not available? I don't know about these devices, so I suppose both are fine but worth giving a thought anyway.